File: solutions_1-10/Euler_9.py
## find the only pythagorean triplet that a + b + c = 1000 ##

import math
import timeit

start = timeit.default_timer()

def euler_9():
    triplet_range = range(1, 3000)
    b = 0
    for a in triplet_range:
        b = triplet_range.index(a) + 1
        while b < len(triplet_range):
            if a + b + (math.sqrt(a**2 + b**2)) == 1000:
                return ("Answer = %s | Triplet: A=%s, B=%s, C=%s" %  
                        (a * b * (math.sqrt(a**2 + b**2)), a, b, (math.sqrt(a**2 + b**2))))
            if a + b + (math.sqrt(a**2 + b**2)) > 1000:
                break
            b += 1

print euler_9()
stop = timeit.default_timer()
print 'Time: %s' % str(stop - start)

File: Java/java_solutions_1-10/Euler_9.java
/*
/   Project Euler problem #9 Special pythagorean triplet
/
/   There exists exactly one pythagorean triplet for which a + b + c = 1000
/   Find the product of abc.
*/

import static java.lang.Math.sqrt;

public class Euler_9
{
    public static long findTriplet(int a, int b)
    {
        int c = (a * a) + (b * b);
        if (sqrt(c) + a + b == 1000) { 
            long triplet = (int) sqrt(c) * a * b;
            return triplet;
        }
        return 0;
    }

    public static void main(String[] args)
    {
        long start = System.nanoTime();
        long triplet = 0;
        long current = 0;
        for (int a = 1; a < 1000; a++) { 
            for (int b = 1; b < 1000; b++) {
                current = findTriplet(a, b);
                if (current > 0) {
                    triplet = current;
                    break;
                }
            }
            if (triplet != 0) {
                break;
            }
        }

        System.out.println("Answer: " + triplet);
        long stop = System.nanoTime();
        System.out.printf("Time: %.4f\n", ((float) stop - start) / 1_000_000_000);
    }
}   